includes/class-coblocks-block-assets.php

```diff
@@ -456,90 +445,139 @@ public function frontend_scripts() {
+		// Post Carousel block script.
+		wp_register_script(
+			'coblocks-post-carousel',
+			$this->assets_dir . 'coblocks-post-carousel.js',
+			array(),
+			COBLOCKS_VERSION,
+			true
+		);
 
-			wp_localize_script(
-				'coblocks-post-carousel',
-				'coblocksPostCarousel',
-				array(
-					'carouselPrevButtonAriaLabel' => $previous_aria_label,
-					'carouselNextButtonAriaLabel' => $next_aria_label,
-				)
-			);
-		}
+		wp_localize_script(
+			'coblocks-post-carousel',
+			'coblocksPostCarousel',
+			array(
+				'carouselPrevButtonAriaLabel' => $previous_aria_label,
+				'carouselNextButtonAriaLabel' => $next_aria_label,
+			)
+		);
 
 		// Events block.
-		if ( $this->is_page_gutenberg() || has_block( 'coblocks/events' ) || has_block( 'core/block' ) ) {
-			wp_enqueue_script(
-				'coblocks-events',
-				$dir . 'coblocks-events.js',
-				array(),
-				COBLOCKS_VERSION,
-				true
-			);
+		wp_register_script(
+			'coblocks-events',
+			$this->assets_dir . 'coblocks-events.js',
+			array(),
+			COBLOCKS_VERSION,
+			true
+		);
 
-			wp_localize_script(
-				'coblocks-events',
-				'coblocksEvents',
-				array(
-					'carouselPrevButtonAriaLabel' => $previous_aria_label,
-					'carouselNextButtonAriaLabel' => $next_aria_label,
-				)
-			);
-		}
+		wp_localize_script(
+			'coblocks-events',
+			'coblocksEvents',
+			array(
+				'carouselPrevButtonAriaLabel' => $previous_aria_label,
+				'carouselNextButtonAriaLabel' => $next_aria_label,
+			)
+		);
 
 		// Counter block.
-		if ( $this->is_page_gutenberg() || has_block( 'coblocks/counter' ) || has_block( 'core/block' ) ) {
-			$asset_file = $this->get_asset_file( 'dist/js/coblocks-counter' );
-			wp_enqueue_script(
-				'coblocks-counter-script',
-				$dir . 'coblocks-counter.js',
-				$asset_file['dependencies'],
-				COBLOCKS_VERSION,
-				true
-			);
-		}
+		$asset_file = $this->get_asset_file( 'dist/js/coblocks-counter-script' );
+		wp_register_script(
+			'coblocks-counter-script',
+			$this->assets_dir . 'coblocks-counter-script.js',
+			$asset_file['dependencies'],
+			COBLOCKS_VERSION,
+			true
+		);
 
 		// Services block.
-		if ( $this->is_page_gutenberg() || has_block( 'coblocks/services' ) ) {
-			$asset_file = $this->get_asset_file( 'dist/js/coblocks-services' );
+		$asset_file = $this->get_asset_file( 'dist/js/coblocks-services-script' );
+		wp_register_script(
+			'coblocks-services-script',
+			$this->assets_dir . 'coblocks-services-script.js',
+			$asset_file['dependencies'],
+			COBLOCKS_VERSION,
+			true
+		);
+
+		$this->localize_lightbox_controls();
+
+	}
+
+	/**
+	 * Enqueue specific scripts for the given core blocks.
+	 *
+	 * @param string $block_content The block content to be rendered.
+	 * @param array  $block         The block attributes.
+	 *
+	 * @return string The original block content.
+	 */
+	public function coblocks_enqueue_scripts_for_core_blocks( $block_content, $block ) {
+
+		$block_name       = $block['blockName'];
+		$block_attributes = $block['attrs'];
```

Comment:
I have a warning on my website : 
`Warning: Undefined array key "attrs" in /var/www/html/wp-content/plugins/coblocks/includes/class-coblocks-block-assets.php on line 519`

Reply:
Good call! I've added some bail logic for if the block does not have attributes, or a name defined.